lib/order.rb

```diff
@@ -0,0 +1,43 @@
+require_relative 'menu'
+
+class Order
+
+  attr_reader :order, :amount
+
+  def initialize
+    @order, @amount = [], []
+  end
+
+  def add_to_order menu, dish, quantity
+      fail 'Item not on menu' if menu.dishes.include?(dish) == false
+      amount << quantity
+      order << dish
+  end
+
+  def total_price
+    total, count = 0, 0
+    order.each do |item|
```

Comment:
You could use `each_with_index` wherever you find yourself manually tracking a count